src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -1399,6 +1411,14 @@ void FancyZones::ApplyQuickLayout(int vKey) noexcept
     FancyZonesDataInstance().SaveZoneSettings();
     UpdateZoneSets(writeLock);
 
+    if (!m_windowMoveHandler.IsDragEnabled())
+    {
+        FlashZones(writeLock);
+    }

```

Comment:
We check `m_windowMoveHandler.IsDragEnabled()` in `OnKeyDown()` and based on that we make some assumptions that affect this functionality. I think it would be better to propagate this information instead re-checking this value here.

Reply:
I agree that the design is not great but it's actually important that the condition is satisfied here, when the lock is held.

Since OnKeyDown calls ApplyQuickLayout asynchronously, it's possible that IsDragEnabled will switch in the meantime and we end up flashing zones while they're already shown - which is not a disaster but it won't look pretty.

In OnKeyDown however, IsDragEnabled is only used to determine whether to accept the key press without modifiers, and misidentifying that while in the middle of switching IsDragEnabled on/off won't even be noticeable.

Reply:
I would rather move the logic in `FlashZones()` that already checks for `m_settings->GetSettings()->flashZonesOnQuickSwitch` so at least we have only one place where we check if the flash has to happen.